app/page.js

```diff
@@ -1,41 +1,74 @@
 "use client";
 
-import React, { useState, useEffect, useRef } from "react";
+import React, { useState, useContext } from "react";
 
-import { spawnNode, Network, NodeConfig } from 'lumina-node';
+import { Network, NodeConfig } from "lumina-node";
+
+import { LuminaContext, LuminaContextProvider } from "./lumina";
 
 export default function Home() {
-  let initialized = useRef(false);
-  let [node, setNode] = useState(null);
+  return (
+    <LuminaContextProvider>
+      <Page />
+    </LuminaContextProvider>
+  );
+}
 
-  useEffect(() => {
-    const init = async () => {
-      let node = await spawnNode();
-      setNode(node);
+function Page() {
+  const node = useContext(LuminaContext);
+
+  const [peerTrackerInfo, setPeerTrackerInfo] = useState();
+  const [syncerInfo, setSyncerInfo] = useState();
+
+  const statsUpdateInterval = setInterval(() => {
+    const update = async () => {
+      if (node && (await node.isRunning())) {
+        setPeerTrackerInfo(await node.peerTrackerInfo());
+        setSyncerInfo(await node.syncerInfo());
+      }
     };
-    if (!initialized.current) {
-      initialized.current = true;
-      init();
-    }
-  }, [node]);
+    update();
+  }, 1000);
 
   return (
     <main>
-      <button onClick={async () => {
-        console.log("current: ", node);
-        let result = await node.is_running();
-        console.log("running: ", result);
-      }}>Is running</button>
-      <br />
-      <button onClick={async () => {
-        let config = NodeConfig.default(Network.Mocha);
-        console.log("bootnodes", config.bootnodes);
-        console.log("network", config.network);
-
-        console.log("current: ", node);
-        let result = await node.start(config);
-        console.log("running: ", result);
-      }}>Start</button>
+      <button
+        onClick={async () => {
+          let config = NodeConfig.default(Network.Mocha);
+          console.log("bootnodes", config.bootnodes);
+          console.log("network", config.network);
+
+          console.log("current: ", node);
+          let result = await node.start(config);
+          console.log("running: ", result);
+        }}
+      >
+        Start
+      </button>
```

Comment:
So starting worker is quick enough that you'll probably not be able to click start quick enough for it to be null, but the proper way would be sth along those lines (but I must admit I hate this syntax):

```js
{ node ?
  <button
    ...
  >
    Start
  </button>
  : <span>Loading...</span>
}
```

Reply:
Added loading state :+1: